docs/reference/upgrade/rolling_upgrade.asciidoc

```diff
@@ -11,14 +11,19 @@ We strongly recommend that when you upgrade you divide your cluster's nodes
 into the following two groups and upgrade the groups in this order:
 
 . Nodes that are not <<master-node,master-eligible>>. You can retrieve a list
-of these nodes with `GET /_nodes/_all,master:false` or by finding all the nodes
-configured with `node.master: false`.
-.. If you are using data tiers, you should
-upgrade the nodes by tier, completing one tier at a time, in this order: frozen, cold, warm and finally hot tier.
-This is to ensure ILM can continue to move between the phases and ensure version
-compatibility.
-.. If you are not using data tiers, you may upgrade the nodes within the group in any order.
-
+of these nodes with `GET /_nodes/_all,master:false/_none` or by finding all the
+nodes configured with `node.master: false`.
+
+.. If you are using data tiers, or a hot-warm-cold architecture based on node 
+attributes, you should upgrade the nodes tier-by-tier, completing the upgrade
+of each tier before starting the next. Upgrade the frozen tier first, then the
+cold tier, then the warm tier, and finally the hot tier.  This is to ensure ILM
+can continue to move between the phases and ensure version
+compatibility. You can get the list of nodes from specific tiers by `GET 
+/_nodes/data_frozen:true/_none`, `GET /_nodes/data_cold:true/_none` etc.
```

Comment:
Unfortunately a line break in the middle of `preformatted text` gets copied into the output:

<img width="707" alt="image" src="https://user-images.githubusercontent.com/5058284/144389829-e6859932-59dc-4a52-b172-d84d4fafa089.png">

It needs to be this instead:

```suggestion
compatibility. You can get the list of nodes from specific tiers by
`GET /_nodes/data_frozen:true/_none`, `GET /_nodes/data_cold:true/_none` etc.
```